plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryMetadata.java

```diff
@@ -104,7 +105,9 @@ public BigQueryMetadata(BigQueryClient bigQueryClient, BigQueryConfig config)
     public List<SchemaTableName> listTables(ConnectorSession session, Optional<String> schemaName)
     {
         log.debug("listTables(session=%s, schemaName=%s)", session, schemaName);
-        Optional<String> remoteSchema = schemaName.map(schema -> bigQueryClient.toRemoteDatasetName(projectId, schema));
+        Optional<String> remoteSchema = schemaName.map(schema -> bigQueryClient.toRemoteDataset(projectId, schema)
+                .map(RemoteDatabaseObject::getAnyRemoteName)
```

Comment:
shouldn't we filter out the ambiguous names here?
Listing way confuse tools like tableau at.al. If they try to get details of table for which conflicting tables exist it would get an error.

Reply:
Don't we consider tables as ambiguous only when their fully qualified name is ambiguous? e.g. are `foo.my_table` and `FOO.your_table` ambiguous? I can't think of any scenario where allowing these two tables in `listTables` output will cause failures.

Also, fully ambiguous tables are handled here by collecting the `SchemaTableName` into a set. Note that `SchemaTableName` lowecases everything on the way in so `FOO.my_table` and `foo.My_Table` will be collapsed into a single entry in `listTables` output.

`listSchemas` behaves similarly because the engine collects `listSchemas` into a set after lowercasing all the entries.

Reply:
Hmmmm, what I'm describing (i.e. what exists today) seems to be not a good experience. We list `FOO.your_table` and `foo.my_table` but without any actual way to use either of those tables (we'll fail when trying to get a table handle) because we can no longer know from `FOO.your_table` if we mean the `foo` or `FOO` schema.

I understand now, will filter here in `listTables`, in `listSchemas` and in `listTableColumns`.